pkg/controller/handlers/toolreference/toolreference.go

```diff
@@ -68,94 +71,26 @@ func New(gptClient *gptscript.GPTScript,
-		return false
-	}
-	return tool.Name != "" && (tool.Type == "" || tool.Type == "tool")
-}
-
 func (h *Handler) toolsToToolReferences(ctx context.Context, toolType types.ToolReferenceType, registryURL string, entries map[string]indexEntry) (result []client.Object) {
-	annotations := map[string]string{
-		"obot.obot.ai/timestamp": time.Now().String(),
-	}
 	for name, entry := range entries {
 		if ref, ok := strings.CutPrefix(entry.Reference, "./"); ok {
 			entry.Reference = registryURL + "/" + ref
 		}
+		if !h.supportDocker && name == system.ShellTool {
+			continue
+		}
 
-		if entry.All {
-			prg, err := h.gptClient.LoadFile(ctx, "* from "+entry.Reference)
-			if err != nil {
-				log.Errorf("Failed to load tool %s: %v", entry.Reference, err)
-				continue
-			}
+		toolRefs, err := tools.ResolveToolReferences(ctx, h.gptClient, name, entry.Reference, true, toolType)
+		if err != nil {
+			log.Errorf("Failed to resolve tool references for %s: %v", entry.Reference, err)
+			continue
+		}
 
-			tool := prg.ToolSet[prg.EntryToolID]
-			if isValidTool(tool) {
-				toolName := tool.Name
-				if tool.MetaData["bundle"] == "true" {
-					toolName = "bundle"
-				}
-				result = append(result, &v1.ToolReference{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:        normalize(name, toolName),
-						Namespace:   system.DefaultNamespace,
-						Finalizers:  []string{v1.ToolReferenceFinalizer},
-						Annotations: annotations,
-					},
-					Spec: v1.ToolReferenceSpec{
-						Type:      toolType,
-						Reference: entry.Reference,
-						Builtin:   true,
-					},
-				})
-			}
-			for _, peerToolID := range tool.LocalTools {
-				// If this is the entry tool, then we already added it or skipped it above.
-				if peerToolID == prg.EntryToolID {
-					continue
-				}
-
-				peerTool := prg.ToolSet[peerToolID]
-				if isValidTool(peerTool) {
-					toolName := peerTool.Name
-					if peerTool.MetaData["bundle"] == "true" {
-						toolName += "-bundle"
-					}
-					result = append(result, &v1.ToolReference{
-						ObjectMeta: metav1.ObjectMeta{
-							Name:        normalize(name, toolName),
-							Namespace:   system.DefaultNamespace,
-							Finalizers:  []string{v1.ToolReferenceFinalizer},
-							Annotations: annotations,
-						},
-						Spec: v1.ToolReferenceSpec{
-							Type:      toolType,
-							Reference: fmt.Sprintf("%s from %s", peerTool.Name, entry.Reference),
-							Builtin:   true,
-						},
-					})
-				}
-			}
-		} else {
-			if !h.supportDocker && name == system.ShellTool {
-				continue
-			}
-			result = append(result, &v1.ToolReference{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:        name,
-					Namespace:   system.DefaultNamespace,
-					Finalizers:  []string{v1.ToolReferenceFinalizer},
-					Annotations: annotations,
-				},
-				Spec: v1.ToolReferenceSpec{
-					Type:      toolType,
-					Reference: entry.Reference,
-					Builtin:   true,
-				},
-			})
+		if entry.NameOverride != "" {
```

Comment:
If I understand this correctly, I can override the top-level name, but not the name of the other tools. So, if I override the name of the time bundle to `the-time`, then all the "sub" tools would have names like `time-sub-tool`, but the top-level tool would be called `the-time`?

Is this right? Does it make sense to override the names of the sub tools too?

Reply:
Hmm, good point. i didn't think of that. Will change it to match the same logic